Update Debian dependencies, fixes Ubuntu Groovy

mfonville opened the pull request:
Atom is not install-able on Ubuntu Groovy because of outdated dependencies as described in #21422 
This fixes and updates Atom dependencies after investigation with @DeeDeeG 
Some dependencies are not required by Electron anymore, some packages are unavailable/outdated and need alternatives
Some dependencies are less strict and can be a suggestion or recommendation. For details see bugreport #21422

mfonville committed:
Update Debian dependencies, fixes Ubuntu Groovy

Atom is not install-able on Ubuntu Groovy because of outdated dependencies as described in #21422 
This fixes and updates Atom dependencies after investigation with @DeeDeeG 
Some dependencies are not required by Electron anymore, some packages are unavailable/outdated and need alternatives
Some dependencies are less strict and can be a suggestion or recommendation. For details see bugreport #21422

mfonville committed:
Update control.in

Keep libsecret-1-0 in suggests

mfonville committed:
Add gnome-keyring

`gnome-keyring` is necessary to run the secret service itself. A KDE-equivalent is not (yet) available. Without its daemon service, libsecret cannot store any secrets.
Put it in recommends so that it is installed by default (majority of use-cases)

mfonville committed:
Keep libxkbfile1 dependency

ldd doesn't see libxkbfile1 as dependency, but when typing in the Atom-editor, missing this library spawns an error

mfonville commented:
Ok, I am happy and certain about these dependencies now.

@darangi @sadick254 is there a possibility to get this into master and get potential feedback from nightlies testing?

Also I'd like to give note that Ubuntu 20.10 will be released in 4 weeks, and this PR is necessary to make Atom install-able.

DeeDeeG commented:
Here's the official documentation of the release schedule, in case that's helpful: https://discourse.ubuntu.com/t/groovy-gorilla-release-schedule/15531

Ubuntu 20.10 comes out October 22.

darangi commented:
Thanks for the additional info @DeeDeeG . @mfonville, I will add this to our radar and the team will look into it soonest.

mfonville committed:
Updated dependencies for trash-handling

According to findings of @DeeDeeG dependencies for trash-handling should be even more limited: https://github.com/atom/atom/issues/21422#issuecomment-703132449 because `ELECTRON_TRASH` variable would otherwise have to be set.

sadick254 reviewed (approved):
This looks great. Tested on Ubuntu Groovy. 👍
